web-services-app/src/test/java/edu/unc/lib/boxc/web/services/processing/ExportCsvServiceTest.java

```diff
@@ -0,0 +1,167 @@
+        exportCsvService.setQueryLayer(solrQueryLayerService);
+        exportCsvService.setRepositoryObjectLoader(repositoryObjectLoader);
+        exportCsvService.setBaseUrl(BASE_URL);
+        resultList = new ArrayList<>();
+        outputStream = new ByteArrayOutputStream();
+
+        when(agent.getPrincipals()).thenReturn(principals);
+        when(response.getResultList()).thenReturn(resultList);
+    }
+
+    @AfterEach
+    void closeService() throws Exception {
+        closeable.close();
+    }
+
+    @Test
+    public void testStreamCsv() {
+        var searchResponse = mock(SearchResultResponse.class);
+        var facet = mock(CutoffFacet.class);
+
+        when(solrQueryLayerService.addSelectedContainer(
+                any(), any(), anyBoolean(), any())).thenReturn(object);
+        when(solrQueryLayerService.getSearchResults(any())).thenReturn(searchResponse);
+
+        when(object.getResourceType()).thenReturn(ResourceType.Folder.name());
+        when(object.getId()).thenReturn(pid.getId());
+        when(object.getTitle()).thenReturn("a good title");
+        when(object.getAncestorNames()).thenReturn("name");
+        when(object.getAncestorPathFacet()).thenReturn(facet);
+        when(object.getRoleGroup()).thenReturn(Arrays.asList("curator|admin", "patron|public"));
+        when(object.getContentStatus()).thenReturn(List.of(FacetConstants.CONTENT_NOT_DESCRIBED));
+        when(facet.getHighestTier()).thenReturn(1);
+
+        exportCsvService.streamCsv(pids, agent, outputStream);
+        verify(accessControlService).assertHasAccess(
+                any(), eq(pid), eq(principals), eq(viewHidden));
+        verify(childrenCountService).addChildrenCounts(any(), any());
+        var result = outputStream.toString(StandardCharsets.UTF_8);
+        assertFalse(result.isBlank());
+    }
+
+    @Test
+    public void testStreamCsvFileObject() {
+        var searchResponse = mock(SearchResultResponse.class);
+        var facet = mock(CutoffFacet.class);
+        var facetNode = mock(CutoffFacetNode.class);
+
+        when(solrQueryLayerService.addSelectedContainer(
+                any(), any(), anyBoolean(), any())).thenReturn(object);
+        when(solrQueryLayerService.getSearchResults(any())).thenReturn(searchResponse);
+        when(object.getResourceType()).thenReturn(ResourceType.File.name());
+        when(object.getId()).thenReturn(pid.getId());
+        when(object.getTitle()).thenReturn("a good title");
+        when(object.getAncestorNames()).thenReturn("/Root/Unit/Collection/Folder/Work/File");
+        when(object.getContentStatus()).thenReturn(List.of(FacetConstants.CONTENT_DESCRIBED));
+        when(object.getAncestorPathFacet()).thenReturn(facet);
+        when(facet.getHighestTier()).thenReturn(1);
+        when(facet.getHighestTierNode()).thenReturn(facetNode);
+        when(facetNode.getSearchKey()).thenReturn(WORK_ID);
+        exportCsvService.streamCsv(pids, agent, outputStream);
```

Comment:
We should probably assert something about this, other than that it runs without errors. We could do similar checks to what are in `testStreamCsv`, stating `never()` for cases where it should not be calling methods, like counting children I think.

Reply:
good point!